WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModel.swift

```diff
@@ -236,11 +240,18 @@ extension ProductFormViewModel {
 // MARK: Remote actions
 //
 extension ProductFormViewModel {
-    func updateProductRemotely(onCompletion: @escaping (Result<EditableProductModel, ProductUpdateError>) -> Void) {
+    func saveProductRemotely(status: ProductStatus?, onCompletion: @escaping (Result<EditableProductModel, ProductUpdateError>) -> Void) {
+        let productModelToSave: EditableProductModel
+        if let status = status, status != product.status {
+            let productWithStatusUpdated = product.product.copy(statusKey: status.rawValue)
+            productModelToSave = EditableProductModel(product: productWithStatusUpdated)
+        } else {
+            productModelToSave = product
+        }
```

Comment:
nit: I'd like to write this kind of code as an inline function
```swift
let productModelToSave: EditableProductModel = {
    guard let status = status, status != product.status else {
        return product
    }
    let productWithStatusUpdated = product.product.copy(statusKey: status.rawValue)
    return EditableProductModel(product: productWithStatusUpdated)
}
```
So it's clear that the variables and code created are limited to the  `productModelToSave` scope

Reply:
Nice it looks much better! Updated in https://github.com/woocommerce/woocommerce-ios/pull/2859/commits/22603684e69a1dd30c302619734d5608358e710a